plugins/inputs/logstash/logstash.go

```diff
@@ -0,0 +1,290 @@
+type CommonData struct {
+	Host        string `json:"host"`
+	Version     string `json:"version"`
+	HTTPAddress string `json:"http_address"`
+	ID          string `json:"id"`
+	Name        string `json:"name"`
+}
+
+// JVMStats data structure
+type JVMStats struct {
+	CommonData CommonData
+	JVM        interface{} `json:"jvm"`
+}
+
+//ProcessStats data structure
+type ProcessStats struct {
+	CommonData CommonData
+	Process    interface{} `json:"process"`
+}
+
+//PluginEvents data structure
+type PluginEvents struct {
+	DurationInMillis float64 `json:"duration_in_millis"`
+	In               float64 `json:"in"`
+	Out              float64 `json:"out"`
+}
+
+//Plugin data structure
+type Plugin struct {
+	ID     string       `json:"id"`
+	Events PluginEvents `json:"events"`
+	Name   string       `json:"name"`
+}
+
+//PipelinePlugins data structure
+type PipelinePlugins struct {
+	Inputs  []Plugin `json:"inputs"`
+	Filters []Plugin `json:"filters"`
+	Outputs []Plugin `json:"outputs"`
+}
+
+//PipelineQueue data structure
+type PipelineQueue struct {
+	Events   float64     `json:"events"`
+	Qtype    string      `json:"type"`
+	Capacity interface{} `json:"capacity"`
+	Data     interface{} `json:"data"`
+}
+
+//Pipeline data structure
+type Pipeline struct {
+	Events  interface{}     `json:"events"`
+	Plugins PipelinePlugins `json:"plugins"`
+	Reloads interface{}     `json:"reloads"`
+	Queue   PipelineQueue   `json:"queue"`
+}
+
+//PipelineStats data structure
+type PipelineStats struct {
+	CommonData CommonData
+	Pipeline   Pipeline `json:"pipeline"`
+}
+
+//Description returns short info about plugin
+func (l *Logstash) Description() string { return "Read metricts exposed by Logstash" }
+
+//SampleConfig returns details how to configure plugin
+func (l *Logstash) SampleConfig() string { return sampleConfig }
+
+//createHttpClient create clients to access API
+func (l *Logstash) createHTTPClient() (*http.Client, error) {
+
+	tr := &http.Transport{
+		ResponseHeaderTimeout: time.Duration(3 * time.Second),
+	}
+
+	client := &http.Client{
+		Transport: tr,
+		Timeout:   time.Duration(4 * time.Second),
+	}
+
+	return client, nil
+}
+
+func (l *Logstash) gatherJSONData(url string, v interface{}) error {
+
+	r, err := l.client.Get(url)
+	if err != nil {
+		return err
+	}
+
+	defer r.Body.Close()
+
+	if r.StatusCode != http.StatusOK {
+		return fmt.Errorf("Logstash: API responded with status-code %d, expected %d",
+			r.StatusCode, http.StatusOK)
+	}
+
+	if err = json.NewDecoder(r.Body).Decode(v); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (l *Logstash) gatherJVMStats(url string, acc telegraf.Accumulator) error {
+	JVMStats := &JVMStats{}
+	if err := l.gatherJSONData(url, JVMStats); err != nil {
+		return err
+	}
+
+	tags := map[string]string{
+		"node_id": JVMStats.CommonData.ID,
+	}
+
+	stats := map[string]interface{}{
+		"jvm": JVMStats.JVM,
+	}
+
+	now := time.Now()
+	for p, s := range stats {
```

Comment:
This is always the same map with a static name, so there is no need for a loop.  Same in the other gather functions.